awx/playbooks/project_update.yml

```diff
@@ -26,127 +26,127 @@
   name: Update source tree if necessary
   tasks:
 
-    - name: delete project directory before update
-      command: "find -delete"  # volume mounted, cannot delete folder itself
+    - name: Delete project directory before update
+      ansible.builtin.command: "find -delete"  # volume mounted, cannot delete folder itself
       args:
         chdir: "{{ project_path }}"
       tags:
         - delete
+      changed_when: false
```

Comment:
This isn't really correct. When the project folder contents are deleted, something _is_ changed. If you're giving a value to changed_when, `true` would be a better shot at being correct. What if you gave changed=true and wrote another expression to skip the task where there are no contents inside the folder? That is the only way I can see making this correct.

Reply:
```
    - name: Detect if project directory is empty
      ansible.builtin.stat:
        path: "{{ project_path }}"
      register: project_dir
      tags:
        - delete

    - name: Delete project directory before update if not empty
      ansible.builtin.command: "find -delete"  # volume mounted, cannot delete folder itself
      args:
        chdir: "{{ project_path }}"
      tags:
        - delete
      changed_when: true
      when: project_dir.stat.exists and project_dir.stat.isdir and project_dir.stat.nlink > 2
 ```
 
 this?

Reply:
i changed it to true is that good enough

Reply:
> i changed it to true is that good enough


It's not accurate:

```
[rick@fedora asdf]$ find . -delete -print
.
[rick@fedora asdf]$ find . -delete -print
.
```

Why should changed be true the second time?

Reply:
I'm worried about performance implications of any version of `-print` because it prints every file. If this is recorded in the event data (as stdout) then that is the exact thing we got ourselves in trouble with when we used the Ansible built-in module (I think file) for this purpose. Once you get to "large" repo sizes (realistically, any major project out there) then the sheer number of files becomes a problem, particularly when the .git folder is included. This kind of situation would make the page un-navigable in the UI and cause a mess of other problems.

You know, we don't even really _need_ to make this determination in the playbook. Since the folder must be on the control node, we could determine in the task code whether there is anything to be deleted. You could even pass in a variable _just_ for the somewhat ridiculous purpose of setting the changed status. This would be preferable to risking performance regressions.

Reply:
> If this is recorded in the event data (as stdout) then that is the exact thing we got ourselves in trouble with when we used the Ansible built-in module (I think file) for this purpose.

Okay, what about:

```yaml
shell: find . -delete -print | head -2`
register: reg
changed_when: reg.stdout_lines | length > 1
```

Reply:
yeah, that should be fine, because the bash commands will behave extremely efficiently in this case.

Reply:
@TheRealHaoLiu if you go this route, note you'll need shell and not command

Reply:
updated as per @relrod's suggestion please re-review